src/modules/esl-media/core/esl-media.less

```diff
@@ -50,5 +56,29 @@
         min-height: 1px;
       }
     }
+
+    &[wide] {
+      &[fill-mode='inscribe'] .esl-media-inner {
+        height: 100%;
+        width: auto;
+      }
+  
+      &[fill-mode='cover'] .esl-media-inner {
+        height: auto;
+        width: 100%;
+      }  
+    }
+
+    &:not([wide])  {
+      &[fill-mode='cover'] .esl-media-inner {
+        height: 100%;
+        width: auto;
+      }
+  
+      &[fill-mode='inscribe'] .esl-media-inner {
+        height: auto;
+        width: 100%;
+      }  
+    }
   }
```

Comment:
```suggestion

    &:is([fill-mode='cover'], [wide][fill-mode='inscribe']) .esl-media-inner {
      width: auto;
      height: 100%;
    }
    &:is([fill-mode='inscribe'], [wide][fill-mode='cover']) .esl-media-inner {
      width: 100%;
      height: auto;
    }
```

Reply:
> 

This suggestion is not logically correct.
For example: our video has wide aspect ratio and we want fill-mode=cover

Second selector [wide][fill-mode='cover']) fits since we have both attribute present.
First selector [fill-mode='cover'] fits since we have cover attribute present

First selector wins since it's first in the stylesheet file, but it wasn't supposed too. My suggestion is uglier, but it works

`
    &:not([wide])[fill-mode='cover'] .esl-media-inner, &[wide][fill-mode='inscribe'] .esl-media-inner {
      width: auto;
      height: 100%;
    }

    &:not([wide])[fill-mode='inscribe'] .esl-media-inner, &[wide][fill-mode='cover'] .esl-media-inner {
      width: 100%;
      height: auto;
    }

`

Please let me know if you have any other suggestions